experimenter/experimenter/nimbus_ui_new/views.py

```diff
@@ -215,3 +216,25 @@ class MetricsUpdateView(NimbusExperimentViewMixin, RequestFormMixin, UpdateView)
     def form_valid(self, form):
         super().form_valid(form)
         return self.render_to_response(self.get_context_data(form=form))
+
+
+class SubscriptionToggleView(RequestFormMixin, NimbusExperimentViewMixin, UpdateView):
+    def post(self, request, *args, **kwargs):
+        experiment = self.get_object()
+        user = request.user
+        is_subscribed = user in experiment.subscribers.all()
+
+        if is_subscribed:
+            experiment.subscribers.remove(user)
+        else:
+            experiment.subscribers.add(user)
+
+        context = {
+            "experiment": experiment,
+        }
+        html = render_to_string(
+            "nimbus_experiments/subscribers_list.html",
+            context,
+            request=request,
+        )
+        return HttpResponse(html)
```

Comment:
Yeah you can technically do it this way but it makes it inflexible for moving other handling logic around in your views, do you wanna try doing the pattern I showed you where you hand this off to the form?

Reply:
Also if you split this into two separate views, one just for subscribing, one for unsubscribing, then you don't need to do the check here at all.

Reply:
let me try that

Reply:
see-also https://docs.djangoproject.com/en/5.1/topics/http/shortcuts/#django.shortcuts.render which is equivalent to HttpResonse(render_to_string(...))